### DISCUSSION TITLE
OAPI - Processes - how to update the status of a long-running job?

### DISCUSSION THREAD

**Discussion by ricardogsilva** [geopython/pygeoapi#1195] [Q&A]
If I have a process that is called in async fashion and it takes a while to run, I would like to be able to provide regular status updates so that clients would be able to monitor the pace of execution.

This implies that there should be a way for a running job to provide updates on its own status to the pygeoapi server.

Looking through the source code I could not find such a way - I might have looked at the wrong place, so please correct me if my analysis is wrong.

I imagine that the running job would need to be allowed to call the `manager.update_job()` method of the job manager and ask it to update the status. It seems that this is currently not possible, as the job is not given any reference to the job manager.

Is there some other way to let a long-running job provide updates on its status?

**Comment by totycro** [comment]
In the pygeoapi-kubernetes-papermill, this is implemented but also happens totally outside of pygeoapi. Here kubernetes jobs are used as backend to keep track of the jobs, so [jobs themselves actually annotate these kubernetes jobs](https://github.com/eurodatacube/papermill-kubernetes-job-progress/blob/main/papermill_kubernetes_job_progress/__init__.py)  with their progress and the [manager reads it from there](https://github.com/eurodatacube/pygeoapi-kubernetes-papermill/blob/main/pygeoapi_kubernetes_papermill/kubernetes.py#L411).

So this probably doesn't help for other use cases, but since the use cases are very different, it might be ok that each one has their own way of keeping track of the progress.
